src/components/DropdownButton.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useState, useRef, useCallback } from "react";
+import { useOnClickOutside } from "usehooks-ts";
+
+interface DropdownButtonProps {
+  buttonText: string;
+  children: React.ReactNode;
+}
+
+export const DropdownButton = (props: DropdownButtonProps) => {
+  const [isActive, setActive] = useState(false);
+  const toggleActive = () => setActive((x) => !x);
```

Comment:
@Cromeror what do you think? should we use a callback here? In my opinion it is too verbose:

```suggestion
  const toggleActive = useCallback(() => setActive((x) => !x), []);
```

Reply:
I agree, because ToogleActive don't need memorize a callback function, is too flat for this.

Reply:
I think is better call the setActive directly on button click.

Reply:
> > Les comparto como queda funcionando con el hook de click outside 👍 Lo único que tenemos que terminar de validar es que no escucha el click dentro del iframe porque no está dentro del mismo document.
> > Una opción es enviarle a Unlayer a través dela configuración un script que envíe los click hechos y de esa manera escucharlos dentro del document.
> 
> Jeje, estaba viendo que la versión basada en _onBlur_ funcionaría. @GusBaamonde @Cromeror creen que se podría lograr un comportamiento aceptable basado en el onblur en lugar del click outside o que podríamos combinar ambos?

Prefiero siempre combinarlos, pienso que usar el _onClick_ para abrir o cerrarlo y el _onBlur_ para cuando se cambia de focus (con teclado o con mouse) ambos usando el mismo llamado a _setActive()_.